crates/extensions/c8y_firmware_manager/src/tests.rs

```diff
@@ -0,0 +1,501 @@
+use super::*;
+
+use assert_json_diff::assert_json_include;
+
+use mqtt_channel::Topic;
+use serde_json::json;
+use sha256::digest;
+use std::str::from_utf8;
+use std::time::Duration;
+use tedge_actors::Actor;
+
+use tedge_actors::DynError;
+use tedge_actors::ReceiveMessages;
+use tedge_actors::SimpleMessageBox;
+use tedge_actors::SimpleMessageBoxBuilder;
+use tedge_config::IpAddress;
+use tedge_test_utils::fs::TempTedgeDir;
+
+use tokio::time::timeout;
+
+const CHILD_DEVICE_ID: &str = "child-device";
+const FIRMWARE_NAME: &str = "fw-name";
+const FIRMWARE_VERSION: &str = "fw-version";
+const TEDGE_HOST: &str = "127.0.0.1";
+const TEDGE_HTTP_PORT: u16 = 8765;
+const TEST_TIMEOUT_MS: Duration = Duration::from_millis(5000);
+const DEFAULT_REQUEST_TIMEOUT_SEC: Duration = Duration::from_secs(3600);
+
+// TODO: We don't need mockito???
+const DOWNLOAD_URL: &str = "http://test.domain.com";
```

Comment:
It depends on the system under test.

You can:

* either test *only* the firmware actor using a `SimpleMessageBox<C8YRestRequest, C8YRestResult>` to check the messages sent to C8Y and simulates C8Y responses. See [this PR](https://github.com/thin-edge/thin-edge.io/pull/1820/files#diff-3ee2f8a335b7513a8e1bfa3087168f91ad41ff9dc900fe3fe1d83bc0877a3120R762) for an example.
* or test the combination `firmware actor <-> c8y proxy actor <-> http actor`. In that case mockito is required to simulate C8Y REST.

Note that you can the same for MQTT: either use a `SimpleMessageBox<MqttMessage, MqttMessage>` to interact with the firmware actor; or connect and spawn an MQTT actor and then use an `mqtt_tests` broker.

I would take the simplest approach in your case.
* If you re-using tests from the former implementation of the firmware manager => Test the whole chain and use Mockito.
* If you are writing *new* tests then focus on the firmware actor and interact with it using message boxes.